openmc/material.py

```diff
@@ -606,6 +608,294 @@ def get_nuclide_densities(self):
+            None will be returned as the figure and axes have already been
+            generated.
+
+        """
+
+        from matplotlib import pyplot as plt
+
+        E, data = self.calculate_xs(types, temperature, cross_sections)
+
+        if divisor_types:
+            cv.check_length('divisor types', divisor_types, len(types),
+                            len(types))
+            Ediv, data_div = self.calculate_xs(divisor_types, temperature,
+                                               cross_sections)
+
+            # Create a new union grid, interpolate data and data_div on to that
+            # grid, and then do the actual division
+            Enum = E[:]
+            E = np.union1d(Enum, Ediv)
+            data_new = np.zeros((len(types), len(E)))
+
+            for l in range(len(types)):
+                data_new[l, :] = \
+                    np.divide(np.interp(E, Enum, data[l, :]),
+                              np.interp(E, Ediv, data_div[l, :]))
+                if divisor_types[l] != 'unity':
+                    types[l] = types[l] + ' / ' + divisor_types[l]
+            data = data_new
+
+        # Generate the plot
+        if axis is None:
+            fig = plt.figure(**kwargs)
+            ax = fig.add_subplot(111)
+        else:
+            fig = None
+            ax = axis
+        for i in range(len(data)):
+            # Set to loglog or semilogx depending on if we are plotting a data
+            # type which we expect to vary linearly
+            if types[i] in PLOT_TYPES_LINEAR:
+                plot_func = ax.semilogx
+            else:
+                plot_func = ax.loglog
+            if np.sum(data[i, :]) > 0.:
+                plot_func(E, data[i, :], label=types[i])
+
+        ax.set_xlabel('Energy [eV]')
+        if divisor_types:
+            ax.set_ylabel('Macroscopic Data')
+        else:
+            ax.set_ylabel('Macroscopic Cross Section [1/cm]')
+        ax.legend(loc='best')
+        ax.set_xlim(Erange)
+        if self.name is not None:
+            title = 'Macroscopic Cross Section for ' + self.name
+            ax.set_title(title)
+
+        return fig
+
+    def calculate_xs(self, types, temperature=294., cross_sections=None):
+        """Calculates continuous-energy macroscopic cross sections of a
+        requested type
+
+        Parameters
+        ----------
+        types : Iterable of values of PLOT_TYPES
+            The type of cross sections to calculate
+        temperature : float, optional
+            Temperature in Kelvin to plot. If not specified, a default
+            temperature of 294K will be plotted. Note that the nearest
+            temperature in the library for each nuclide will be used as opposed
+            to using any interpolation.
+        cross_sections : str, optional
+            Location of cross_sections.xml file. Default is None.
+
+        Returns
+        -------
+        unionE : numpy.array
```

Comment:
I would prefer something like `energy_grid` that is a little more descriptive and follows the under_line syntax.